### PR TITLE
Add only loading partial NetCDF batches

### PR DESCRIPTION
**Pull request by jacobbieker**
# Pull Request

## Description

Currently the dataloader loads all the NEtCDF files in the batch before then throwing some of them away, possibly. This wastes time and compute. This PR adds loading only a subset of the NetCDF files if desired.

Companion PR to be merged first: https://github.com/openclimatefix/nowcasting_dataset/pull/510

This PR also:

- [x] Fixes an issue with the PV Yield being in the wrong order and not getting the position encoding
- [ ] 

Fixes #

## How Has This Been Tested?

Unit tests

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: tests/test_netcdf_dataset.py [hunk 1]
```diff
@@ -83,6 +83,67 @@ def test_netcdf_dataset_local_using_configuration():
         assert os.path.exists(os.path.join(DATA_PATH, "nwp/000000.nc"))
 
 
+def test_netcdf_dataset_local_using_configuration_subset_of_data_sources():
```
Comment (peterdudfield) [inline tests/test_netcdf_dataset.py:86]: nice

File: tests/test_satflow_dataset.py [hunk 2]
```diff
@@ -152,6 +152,76 @@ def test_satflow_dataset_local_using_configuration_with_position_encoding():
         assert os.path.exists(os.path.join(DATA_PATH, "nwp/000000.nc"))
 
 
+def test_satflow_dataset_local_using_configuration_with_position_encoding_subset_of_sources():
```
Comment (peterdudfield) [inline tests/test_satflow_dataset.py:155]: nice

**Review by peterdudfield — APPROVED** [review]

File: nowcasting_dataloader/datasets.py [hunk 3]
```diff
@@ -68,6 +69,7 @@ def __init__(
             cloud: which cloud is used, can be "gcp", "aws" or "local".
             normalize: normalize the batch data
             add_position_encoding: Whether to add position encoding or not
+            data_sources_names: Names of data sources to load, if not using all of them
```
Comment (JackKelly) [inline nowcasting_dataloader/datasets.py:72]: I might suggest making the docstring a tiny bit more explicit (and below, too, in `SatFlowDataset.__init__`), something like...

```suggestion
            data_sources_names: Names of data sources to load.  Leave as None to load all data sources.
```

**Review by JackKelly — APPROVED** [review]
LGTM!  Thanks for implementing this, and writing tests!  Very nice!

**Pull request merged**